View3D/Utility/GeometryIntersection.cs

```diff
@@ -0,0 +1,181 @@
+                    var dist = intersectionPoint;
+                    if (dist < bestDistance)
+                    {
+                        faceIndex = i;
+                        bestDistance = dist.Value;
+                    }
+                }
+            }
+
+            if (faceIndex == -1)
+                return null;
+
+            face = faceIndex;
+            return bestDistance;
+        }
+
+        public static bool IntersectObject(BoundingFrustum boundingFrustum, MeshObject geometry, Matrix matrix)
+        {
+            for (int i = 0; i < geometry.VertexCount(); i++)
+            {
+                if (boundingFrustum.Contains(Vector3.Transform(geometry.GetVertexById(i), matrix)) != ContainmentType.Disjoint)
+                    return true;
+            }
+
+            return false;
+        }
+
+        public static bool IntersectFaces(BoundingFrustum boundingFrustum, MeshObject geometry, Matrix matrix, out List<int> faces)
+        {
+            faces = new List<int>();
+
+            var indexList = geometry.GetIndexBuffer();
+            var vertList = geometry.GetVertexList();
+
+            var transformedVertList = new Vector3[vertList.Count];
+            for (int i = 0; i < vertList.Count; i++)
+                transformedVertList[i] = Vector3.Transform(vertList[i], matrix);
+
+            for (int i = 0; i < indexList.Count; i += 3)
+            {
+                var index0 = indexList[i + 0];
+                var index1 = indexList[i + 1];
+                var index2 = indexList[i + 2];
+
+                if (boundingFrustum.Contains(transformedVertList[index0]) != ContainmentType.Disjoint)
+                    faces.Add(i);
+                else if (boundingFrustum.Contains(transformedVertList[index1]) != ContainmentType.Disjoint)
+                    faces.Add(i);
+                else if (boundingFrustum.Contains(transformedVertList[index2]) != ContainmentType.Disjoint)
+                    faces.Add(i);
+            }
+
+            if (faces.Count() == 0)
+                faces = null;
+            return faces != null;
+        }
+
+        public static bool IntersectVertices(BoundingFrustum boundingFrustum, MeshObject geometry, Matrix matrix, out List<int> vertices)
+        {
+            vertices = new List<int>();
+
+            for (int i = 0; i < geometry.GetIndexCount(); i++)
+            {
+                var index = geometry.GetIndex(i);
+                
+                if (boundingFrustum.Contains(Vector3.Transform(geometry.GetVertexById(index), matrix)) != ContainmentType.Disjoint)
+                    vertices.Add(index);
+            }
+            vertices = vertices.Distinct().ToList();
+            if (vertices.Count() == 0)
+                vertices = null;
+            return vertices != null;
+        }
+
+        internal static bool IntersectBones(BoundingFrustum boundingFrustum, Rmv2MeshNode sceneNode, GameSkeleton skeleton, Matrix matrix, out List<int> bones)
```

Comment:
If possible, dont use the sceneNode here. 
I want to change/remove that class as its a mess, so I want to avoid it spreading

Reply:
removed in 1b957d5f07fde6d3a956365e7f809b9c7506fb83